src/SonarQube.Client/Api/SonarQubeService.cs

```diff
@@ -0,0 +1,252 @@
+                },
+                token);
+
+        public async Task<IList<SonarQubeProperty>> GetAllPropertiesAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPropertiesRequest, SonarQubeProperty[]>(token);
+
+        public Uri GetProjectDashboardUrl(string projectKey)
+        {
+            EnsureIsConnected();
+
+            const string ProjectDashboardRelativeUrl = "dashboard/index/{0}";
+
+            return new Uri(httpClient.BaseAddress, string.Format(ProjectDashboardRelativeUrl, projectKey));
+        }
+
+        public async Task<SonarQubeQualityProfile> GetQualityProfileAsync(string projectKey, string organizationKey, SonarQubeLanguage language, CancellationToken token)
+        {
+            var qualityProfiles = await InvokeRequestAsync<IGetQualityProfilesRequest, SonarQubeQualityProfile[]>(
+                request =>
+                {
+                    request.ProjectKey = projectKey;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+            // Consider adding the language filter to the request configuration above and removing this line
+            var profilesWithGivenLanguage = qualityProfiles.Where(x => x.Language == language.Key).ToList();
+
+            var qualityProfile = profilesWithGivenLanguage.Count > 1
+                ? profilesWithGivenLanguage.Single(x => x.IsDefault)
+                : profilesWithGivenLanguage.Single();
+
+            var changeLog = await InvokeRequestAsync<IGetQualityProfileChangeLogRequest, DateTime[]>(
+                request =>
+                {
+                    request.Page = 1;
+                    request.PageSize = 1;
+                    request.QualityProfileKey = qualityProfile.Key;
+                    request.QualityProfileName = qualityProfile.Name;
+                    request.LanguageName = language.Key;
+                },
+                token);
+
+            var updatedDate = changeLog.Any() ? changeLog.Single() : qualityProfile.TimeStamp;
+
+            return new SonarQubeQualityProfile(qualityProfile.Key, qualityProfile.Name, qualityProfile.Language,
+                qualityProfile.IsDefault, updatedDate);
+        }
+
+        public async Task<RoslynExportProfileResponse> GetRoslynExportProfileAsync(string qualityProfileName,
+            string organizationKey, SonarQubeLanguage language, CancellationToken token) =>
+            await InvokeRequestAsync<IGetRoslynExportProfileRequest, RoslynExportProfileResponse>(
+                request =>
+                {
+                    request.QualityProfileName = qualityProfileName;
+                    request.LanguageKey = language.Key;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+        public async Task<IList<SonarQubeIssue>> GetSuppressedIssuesAsync(string key, CancellationToken token)
```

Comment:
Can key be null/empty?

Reply:
If it is null or empty we will pass empty query string argument, which is ok, I think. I don't know what SQ will do in this case, though :)